MeetU/MeetU/Scripts/app/meetu-meetup.js

```diff
@@ -14,24 +14,31 @@
             var Userview = $resource('/api/loggedUser');
             var Join = $resource('/api/Joins');
             var CommentView = $resource('/api/Comments/');
+
+            $scope.hasLoaded = false;
+
             //
             //  Queries
             //
-            Meetup.query(function (data) {
+            Meetup.query(function(data) {
                 $scope.meetupViews = data;
-            });
-            Userview.query(function (userViews) {
-                $scope.userId = userViews[0].userId;
-                $scope.userName = userViews[0].userName;
-            });
-            CommentView.query(function (data) {
-                $scope.allCommentViews = data;
+            }).$promise.then(function() {
```

Comment:
I agree your solution about time overhead. I am just wondering if the http request is sent one by one? if it is, that the overhead would not be much different. But anyway, your solution is better, I will do that.